failing unit test: TestUnsetVolunteer

may require #1815 to be merged first

the following test is failing: TestUnsetVolunteer:test_badge_at_con()

```
(fixture: c.AT_THE_CON = True)

        monkeypatch.setattr(Attendee, 'is_new', False)
        a = Attendee(badge_num=1)
        a._misc_adjustments()
        assert not a.checked_in
```

```
FAILED

self = <uber.tests.models.test_attendee.TestUnsetVolunteer object at 0x7f4d4acd2b38>
monkeypatch = <_pytest.monkeypatch.monkeypatch object at 0x7f4d4aa9ab70>
at_con = None

    def test_badge_at_con(self, monkeypatch, at_con):
        a = Attendee()
        a._misc_adjustments()
        assert not a.checked_in

        a = Attendee(badge_num=1)
        a._misc_adjustments()
        assert a.checked_in

        monkeypatch.setattr(Attendee, 'is_new', False)
        a = Attendee(badge_num=1)
        a._misc_adjustments()
>       assert not a.checked_in
E       assert not datetime.datetime(2016, 4, 29, 13, 17, 29, 364947, tzinfo=<UTC>)
E        +  where datetime.datetime(2016, 4, 29, 13, 17, 29, 364947, tzinfo=<UTC>) = <Attendee full_name=' '>.checked_in

sideboard/plugins/uber/uber/tests/models/test_attendee.py:245: AssertionError
```

--- a/uber/models.py
+++ b/uber/models.py
@@ -1193,7 +1193,7 @@ class Attendee(MagModel, TakesPaymentMixin):
             self.paid = c.NEED_NOT_PAY
 
         if c.AT_THE_CON and self.badge_num and not self.checked_in and \
-                (self.is_new or self.badge_type not in c.PREASSIGNED_BADGE_TYPES):
+                self.is_new and self.badge_type not in c.PREASSIGNED_BADGE_TYPES:
             self.checked_in = datetime.now(UTC)
 
         if self.birthdate:

--- a/uber/tests/models/test_attendee.py
+++ b/uber/tests/models/test_attendee.py
@@ -189,10 +189,14 @@ class TestUnsetVolunteer:
         a._misc_adjustments()
         assert a.checked_in
 
+        a = Attendee(badge_num=1, badge_type=c.PREASSIGNED_BADGE_TYPES[0])
+        a._misc_adjustments()
+        assert not a.checked_in
+
         monkeypatch.setattr(Attendee, 'is_new', False)
         a = Attendee(badge_num=1)
         a._misc_adjustments()
-        assert a.checked_in
+        assert not a.checked_in
 
     def test_names(self):
         a = Attendee(first_name='nac', last_name='mac Feegle')

